cli/command/container/run.go

```diff
@@ -228,16 +228,14 @@ func runContainer(ctx context.Context, dockerCli command.Cli, runOpts *runOption
 		}
 	}
 
-	if errCh != nil {
-		if err := <-errCh; err != nil {
-			if _, ok := err.(term.EscapeError); ok {
-				// The user entered the detach escape sequence.
-				return nil
-			}
-
-			logrus.Debugf("Error hijack: %s", err)
-			return err
+	if err := <-errCh; err != nil {
```

Comment:
I went looking if `errCh` should be defined closer to where it's used, but then also noticed the;

```
	if !attach {
		// Make this asynchronous to allow the client to write to stdin before having to read the ID
		...
	}
	if attach {
```

Which could be an if/else. But at that point even started to consider if we should (_not for this PR!_) split the whole function in to `runAttached` / `runDetached` - it seems like there's various code-paths  intertwined, each requiring specific conditions/checks, that perhaps things could be cleaner if they were  separated (at the cost of _some_ duplication).

I haven't tried what that'd look like, but perhaps it's something we could give a try at some point to see how it looks.

Reply:
I for sure agree! We chatted about this privately, but to restate here, I have a branch with a bigger refactor (splitting out the "detached" logic entirely), and these are just smaller, cherry-picked changes from it. These are a bit safer IMO, it's easier to follow these commits, so we can merge this one first and I'll open the larger refactor on top of it.

Reply:
Yup; definitely not needed for this PR!